bjoern/server.c

```diff
@@ -233,12 +264,18 @@ ev_io_on_read(struct ev_loop* mainloop, ev_io* watcher, const int events)
   switch (read_state) {
   case not_yet_done:
     break;
+  case expect_continue:
+    DBG_REQ(request, "pause read, write 100-continue");
+    ev_io_stop(mainloop, &request->ev_watcher);
+    request->current_chunk = _PEP3333_Bytes_FromString(
+      CONTINUE_RESPONSE);
+    start_writing(mainloop, request);
+    break;
   case done:
     DBG_REQ(request, "Stop read watcher, start write watcher");
     ev_io_stop(mainloop, &request->ev_watcher);
-    ev_io_init(&request->ev_watcher, &ev_io_on_write,
-               request->client_fd, EV_WRITE);
-    ev_io_start(mainloop, &request->ev_watcher);
+    clear_expectation_state(request);
```

Comment:
Why do we need this here?

Reply:
This = `clear_expectation_state`

Reply:
There is a condition where `expect_continue` might be `true` but the client decided not to wait (I describe this in the new doc/comment in `on_header_value` in request.c). But I moved this toggle to a more appropriate place (`on_body` in request.c) and it's no longer required here.